jupyterlab_git/tests/test_clone.py

```diff
@@ -32,6 +32,30 @@ async def test_git_clone_success():
             assert {"code": 0, "message": output} == actual_response
 
 
+@pytest.mark.asyncio
+async def test_git_download_success(tmp_path):
+    with patch("os.environ", {"TEST": "test"}):
+        with patch("jupyterlab_git.git.execute") as mock_execute:
+            # Given
+            output = "output"
+            mock_execute.return_value = maybe_future((0, output, "error"))
+
+            # When
+            await Git().clone(
+                path=str(Path("/bin/test_curr_path")), repo_url="ghjkhjkl"
+            )
+
+            # Then
+            mock_execute.assert_called_once_with(
+                ["git", "clone", "ghjkhjkl"],  # to be update
+                cwd=str(Path("/bin") / "test_curr_path"),  # to be udpated
+                env={"TEST": "test", "GIT_TERMINAL_PROMPT": "0"},
+            )
+
+            # Check the `.git` folder has been removed.
+            assert not (str(tmp_path) / ".git").exists()
```

Comment:
@sheezaaziz I forgot that we need to create the folder `repo/.git` when `execute` is called. So this test is gonna be a bit more advance as we need to use the `side_effect` feature of `patch` instead of `return_value` (see [documentation](https://docs.python.org/3/library/unittest.mock.html?highlight=patch#unittest.mock.Mock.side_effect))

Something like that

```suggestion
            git_folder = tmp_path / "dummy-repo" / ".git"
            def create_fake_git_repo(*args, **kwargs):
                git_folder.mkdir(parents=True)
                return maybe_future((0, output, "error"))
            mock_execute.side_effect = create_fake_git_repo

            # When
            await Git().clone(
                path=str(tmp_path), repo_url="ghjkhjkl", None, False)
            )

            # Then
            mock_execute.assert_called_once_with(
                ["git", "clone", "--depth=1", "ghjkhjkl"],  # to be update
                cwd=str(tmp_path),  # to be udpated
                env={"TEST": "test", "GIT_TERMINAL_PROMPT": "0"},
            )

            # Check the repository folder has been created
            assert git_folder.parent.exists()
            # Check the `.git` folder has been removed.
            assert not git_folder.exists()
```